src/main/java/uk/gov/companieshouse/documentsigningrequestconsumer/satisfyitem/SatisfyItemApiPatch.java

```diff
@@ -48,4 +53,22 @@ public void satisfyItem(ServiceParameters parameters, int status, String documen
         logger.info("API returned response: "+ response.getStatusCode(),
             getLogMap(parameters.getData().get(ORDER_ID).toString(), parameters.getData().get(GROUP_ITEM).toString()));
     }
+
+    private String getGroupItemExcludingItemId (final String groupItem)
+    {
+        //
+        // item group is populated by item-group-workflow-api using item->links->self
+        // Correct format check.
+        //
+        if (!URIValidator.validate(PrivateSatisfyItemURIPattern.getCompiledSatisfyItemsURIPattern(), groupItem))
```

Comment:
I was really thinking just checking for the presence of the required `/` would have been sufficient (as we validate the URI a bit further on anyway), however this should work too. Some other minor points, you could have avoided calling `PrivateSatisfyItemURIPattern.getCompiledSatisfyItemsURIPattern()` twice, and perhaps a unit test would have been nice to verify/demonstrate this functionality.

Reply:
Spoke to Wayne about log errors and he liked the idea of some idea as to what the format should look like, so added in the format to the exception message. The tests cover any NonRetryableException as a group, so in the interest of brevity decided to leave as is.